src/main/java/world/inetum/realdolmen/inetumrealJobs/services/UserDetailsImpl.java

```diff
@@ -71,7 +68,7 @@ public String getPassword() {
 
     @Override
     public String getUsername() {
-        return username;
+        return null;
```

Comment:
Shouldn't this return email?

Reply:
Only if we consider that the email is a username but if we do so, the method name has to be changed but since it is an override method we cant. This is why I added the getEmail() method on line 60

Reply:
Fixed